pkg/lib/serverTime.js

```diff
@@ -321,13 +324,181 @@ export function ServerTime() {
+
+        function confline(line, list, name) {
+            const c = list.find(e => e.name == name);
+            if (!c)
+                return line;
+            const line_trim = line.trim();
+            const line_enabled = !line_trim.startsWith("#");
+            if (c.enabled && !line_enabled)
+                return line_trim.substr(1);
+            if (!c.enabled && line_enabled)
+                return "#" + line_trim;
+            return line;
+        }
+
+        return chrony_file.modify(text => {
+            let new_text = "";
+
+            let in_cockpit_section = false;
+            for (const line of text.trimEnd().split("\n")) {
+                if (line == "# Added by Cockpit")
+                    in_cockpit_section = true;
+                else {
+                    const fields = line.split(" ");
+                    const was_in_cockpit_section = in_cockpit_section;
+                    if (fields[0] != "server")
+                        in_cockpit_section = false;
+                    if (!in_cockpit_section && (!was_in_cockpit_section || line != "")) {
+                        if (fields[0] == "pool" || fields[0] == "#pool") {
+                            new_text += confline(line, config.standard, fields[1]) + "\n";
+                        } else if (fields[0] == "server" || fields[0] == "#server") {
+                            new_text += confline(line, config.standard, fields[1]) + "\n";
+                        } else if (fields[0] == "sourcedir" || fields[0] == "#sourcedir") {
+                            new_text += confline(line, config.sources, fields[1]) + "\n";
+                        } else {
+                            new_text += line + "\n";
+                        }
+                    }
+                }
+            }
+
+            const extra = config.extra.filter(e => !!e);
+
+            if (extra.length > 0) {
+                if (!new_text.endsWith("\n"))
+                    new_text += "\n";
+                if (!new_text.endsWith("\n\n"))
+                    new_text += "\n";
+
+                new_text += "# Added by Cockpit\n";
+                for (const e of extra)
+                    new_text += "server " + e + "\n";
+            }
+
+            return new_text;
+        });
+    }
+
+    self.get_ntp_config = function () {
+        /* We support editing the configuration of systemd-timesyncd
+         * or chronyd, by dropping a config file snippet that we
+         * assume full control over.
+         */
+
+        const backend = get_timesync_backend();
+        if (backend == "timesyncd") {
+            return get_ntp_config_timesyncd();
+        } else if (backend == "chronyd") {
+            return get_ntp_config_chronyd();
+        } else {
+            return Promise.resolve({ supported: false });
+        }
+    };
+
+    self.set_ntp_config = function (config) {
+        if (!config || !config.supported)
+            return;
```

Comment:
`!config.supported` is a Should Not Happen™, let's raise a console.error here.

Reply:
Hmm, no, this happens with ntpd, for example, and is not an error.  Cockpit will allow NTP to be enabled, but wont allow any configuration on-top of that.